sdk/servicebus/service-bus/src/constructorHelpers.ts

```diff
@@ -23,8 +24,12 @@ import { SharedKeyCredential } from "./servicebusSharedKeyCredential";
  *       this is used to compute the exponentially increasing delays between retries. Default: 30000 milliseconds.
  *    - `timeoutInMs`: Amount of time in milliseconds to wait before the operation times out. This will trigger a retry if there are any
  *       retry attempts remaining. Minimum value: 60000 milliseconds.
+ * - `requestOptions`   : Options used when creating and sending HTTP requests for this operation. (Not applicable for the pure AMQP operations.)
+ *    - `timeout`   : The number of milliseconds a request can take before automatically being terminated. If provided, this timeout will be passed/used for the auth operations such as obtaining the token while using `@azure/identity` and Azure AD credentials.
```

Comment:
What's the default?

Reply:
I haven't looked at the defaults.
From the discussion with @sadasant here https://github.com/Azure/azure-sdk-for-js/issues/13436, the default was stated as 500ms.

Reply:
We do typically document these here though, right? We did for the above value for timeoutInMs.

Reply:
I was wondering.. since we don't manage the default value, we shouldn't mention it?
What if identity changes the default value? We'll be stale, that's fine?

Reply:
If we think the current value is already problematic then I assume we're going to increase it, right? In which case we own the default value and we can document what it is (it's probably going to be higher than 500ms).

Or is the idea that we leave this setting in place and it just fails and we somehow provide guidance that will lead them to this setting?

Reply:
With regards to identity, the same timeout would be applicable for all kinds of credentials.
I checked the user issue again, the default for the case of Managed Identity credential is 500ms which would have been sufficient for a normal managed identity, not enough for the "aad-pod-identity".

> If we think the current value is already problematic then I assume we're going to increase it, right?

The current value is problematic for the specific case of a specific credential. I'm unsure of how `@azure/identity` would handle it. (@sadasant thoughts?)

And `@azure/identity` is free to pick defaults differently for each of the credentials offered, so we may not be able to mention the defaults for each of them.

> Or is the idea that we leave this setting in place and it just fails and we somehow provide guidance that will lead them to this setting?

Most likely. `getToken` expects a timeout from the client libraries depending on it. With this PR, `service-bus` exposes the control to the users just like the other SDKs.